sqlalchemy_ibmi/base.py

```diff
@@ -592,27 +605,22 @@ def pre_exec(self):
             insert_has_sequence = seq_column is not None
 
             self._select_lastrowid = insert_has_sequence and \
-                not self.compiled.returning and \
-                not self.compiled.inline
+                                     not self.compiled.returning and \
+                                     not self.compiled.inline
 
     def post_exec(self):
-        conn = self.root_connection
+        conn = self.connection()
+        cur = conn.create_cursor()
```

Comment:
I think this was fine as is, since DefaultExecutionContext sets up root_connection: https://github.com/sqlalchemy/sqlalchemy/blob/fa1f67a01e80367d73cf5d1d93b6f7f51dc1746b/lib/sqlalchemy/engine/default.py#L691

It also sets up the cursor: https://github.com/sqlalchemy/sqlalchemy/blob/fa1f67a01e80367d73cf5d1d93b6f7f51dc1746b/lib/sqlalchemy/engine/default.py#L715

Reply:
Indeed, SQLAlchemy's built-in mssql execution context does exactly the same as the original code, other than the SQL being different: https://github.com/sqlalchemy/sqlalchemy/blob/25b44105269ee38778464c28e5869a742f9cc407/lib/sqlalchemy/dialects/mssql/base.py#L1476-L1485